src/Archives/AdaptiveHuffmanTree.cpp

```diff
@@ -127,14 +124,23 @@ namespace Archives
 	// Raise exception if nodeIndex is out of range
 	void AdaptiveHuffmanTree::VerifyValidNodeIndex(NodeIndex nodeIndex)
 	{
-		// Check that the nodeIndex is in range
 		if (nodeIndex >= nodeCount)
 		{
 			throw std::runtime_error("AdaptiveHuffmanTree NodeIndex of " + std::to_string(nodeIndex)
 				+ " is out of range " + std::to_string(nodeCount));
 		}
 	}
 
+	// Raise exception if code is out of range
+	void AdaptiveHuffmanTree::VerifyValidDataValue(DataValue code)
```

Comment:
This function name could probably benefit from shortening. Maybe 'VerifyData'. If that seems too short, 'VerifyDataValue'.

Reply:
I chose the name to closely match the `VerifyValidNodeIndex` method that already exists.

I'm not entirely in love with those two names either. I thought about shortening them, but also considered if they need to actually be longer. I think it's ok to use long names here. They're private members so their scope is limited anyway.

In terms of writing "verify" methods that throw exceptions, I think it's good practice for the name to somehow indicate what they are verifying, or under what conditions they throw. The "verify" word alone doesn't describe this, and neither does a noun indicating the type it works on. Hence I don't think "Verify<NounPhrase>" is good pattern for naming. As such, I'd like to avoid using "VerifyData", or "VerifyDataValue".

In this case, we are verifying the range of values for that type falls within the expected range. Perhaps a better name is `VerifyDataValueInRange`, though that sounds almost like it expects a range argument.

Reply:
I think my issue is the word Valid by itself is too generic and should just be removed if left by itself. Maybe like you mentioned it would be better just to be more descriptive then. 

Maybe VerifyDataRange or VerifyDataValueRange? I think these would be more descriptive of what part of DataValue is being verified and the names are not much longer. 

After thinking about it some more, part of the problem might be because DataValue consists of 2 generic terms. It isn't clear that DataValue belongs attached to a node. Maybe NodeData would be a better name as it would indicate the data came from a node? 

```
void AdaptiveHuffmanTree::VerifyNodeDataRange(NodeData code);
void AdaptiveHuffmanTree::VerifyNodeIndexRange(NodeIndex nodeIndex);
```

I think these could be shortened to below, but maybe this becomes too terse?

```
void AdaptiveHuffmanTree::VerifyDataRange(NodeData code);
void AdaptiveHuffmanTree::VerifyIndexRange(NodeIndex nodeIndex)
```

Using the name NodeData would go well with the following function as well:
```
NodeData GetNodeData(NodeIndex nodeIndex);
```

Hope this helps?

-Brett

Reply:
I think we're on the same page here. The problem isn't the length, but rather it was a bad name. Coincidentally, I was also thinking about renaming `DataValue` to `NodeData`.

Ok, I'll do some renaming then.